### ISSUE TITLE
Show length of a task in the calendar

### ISSUE THREAD

**Issue by xanxanxan** [nextcloud/calendar#4752]
### Is your feature request related to a problem? Please describe.

I love that the web interface calendar shows calendar entries alongside tasks. I find that so useful and so far I found no other calendar app that did this. Thank you very much for this great feature!

However, each task in the calendar has a fixed length. It takes out the space of half an hour. But not every task is done in half an hour.

### Describe the solution you'd like

To help me with my day planning and productivity, the calendar app could allow me to define how long I think I will need to complete a task. Isn’t that what 'start date' and 'due date' are for?

I’d like the calendar app to consider both these properties and show the box accordingly. I’d also like to be able to drag and drop the task to a different day or time and tweak it’s length with my mouse.

### Describe alternatives you've considered

Setting 'start date' in the tasks app has no effect on how the task is shown in the calendar app.

I can augment the length of a task in the calendar app and for the moment, it is shown as intended. But as soon as I switch to another week and then go back to the week the task is in, it is back to a length of half an hour.

**Comment by oktayacikalin** [comment]
I think, a task can have a start and due date. But what I feel is missing here is something like a duration (0.25h, 0.5h, 1h, 2h, 4h, 6h, 8h). The calendar should then show the task at the start datetime, and the stop datetime should reflect the duration.

**Comment by miaulalala** [comment]
The VTODO components are written/editable from the Tasks app, so the ticket should be openend with them, we only support displaying them.

Before I transfer the issue, could you please export a VTODO for me and post it?

https://icalendar.org/iCalendar-RFC-5545/3-6-2-to-do-component.html for the specs. Due / duration is supported.

**Comment by tcitworld** [comment]
The issue is that some tasks can sometimes go on for several days (a due date might not mean you're working on it full time), filling the calendar UI.

**Comment by oktayacikalin** [comment]
Right, I also think that the due date is not the same as the duration. What about asking the Task team to implement a real duration item, which would then use the stop datetime to reflect this? And they should make it all-day if it spans multiple days.

**Comment by rokejulianlockhart** [comment]
I always consider the start and end dates to mean the duration.
